proof/invariant-abstract/AARCH64/ArchVCPUDomain_AI.thy

```diff
@@ -0,0 +1,605 @@
+(*
+ * Copyright 2024, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+theory ArchVCPUDomain_AI
+imports ArchVCPU_AI
+begin
+
+context Arch begin arch_global_naming
+
+section \<open>cur_vcpu_in_cur_domain\<close>
+\<comment> \<open>FIXME: @{term cur_vcpu} could be reworked so that it could be reused here\<close>
+definition cur_vcpu_tcb :: "'z::state_ext state \<Rightarrow> obj_ref option" where
+  "cur_vcpu_tcb s \<equiv>
+     case arm_current_vcpu (arch_state s) of
+       Some (v, _) \<Rightarrow> vcpu_tcbs_of s v
+     | None \<Rightarrow> None"
```

Comment:
Is that FIXME something we actually want to/should do? I think we looked at it in the past, and it doesn't quite work out without changing lots of things. If that is the case, we should probably remove the FIXME, and turn it into a comment that says that we have looked at it, so we don't repeat the process.